### DISCUSSION TITLE
外部サーバーの画像を指定した場合、camo.qiitausercontent.comの画像キャッシュ生成に失敗する

### DISCUSSION THREAD

**Discussion by MasatoMakino** [increments/qiita-discussions#233] [要望・アイデア・不具合]
# 発生する症状

Qiitaの記事を投稿 / 編集する際、[画像リンク記法](https://qiita.com/Qiita/items/c686397e4a0f4f11683d#images---%E7%94%BB%E5%83%8F%E5%9F%8B%E3%82%81%E8%BE%BC%E3%81%BF)のリンク先に https://qiita-image-store.s3.ap-northeast-1.amazonaws.com 以外の外部サーバーを指定すると、camo.qiitausercontent.comサーバーがNot Foundを返します。

# 発生する記事とリビジョン

以下の記事を更新する際に、上記の症状が発生しました。
https://qiita.com/masato_makino/items/dcc2ba13d28a5a6abab1

2021年10月10日 14時25分 (JST)の[リビジョン](https://qiita.com/masato_makino/items/dcc2ba13d28a5a6abab1/revisions/5)では問題は発生せず、
2022年05月28日 10時26分 (JST)の[リビジョン](https://qiita.com/masato_makino/items/dcc2ba13d28a5a6abab1/revisions/6)の更新を開始した際に、Webブラウザ上のマークダウンエディタで画像が表示されなくなりました。

# 確認した事項

camo.qiitausercontent.comサーバーは、外部サーバーの画像を指定した場合でもユニークなURLを発行しているようです。

例えば上記の記事の場合、
https://gam0022.net/images/posts/2018-12-24-vscode-glslsandbox/traveler2-win.png
への画像リンクを記述した場合
https://camo.qiitausercontent.com/6b28db9a72ea1e920c237e238c2d21efa8132e4f/68747470733a2f2f67616d303032322e6e65742f696d616765732f706f7374732f323031382d31322d32342d7673636f64652d676c736c73616e64626f782f74726176656c6572322d77696e2e706e673f6c6566
というユニークなURLが生成されていました。

しかしこのURLにアクセスしても、Not Foundというレスポンスが返ってきます。

# 影響範囲

過去記事を再編集する場合、 https://qiita-image-store.s3.ap-northeast-1.amazonaws.com 以外にアップロードした画像すべてが影響を受けます。過去記事を更新しない限りこの問題の影響は受けません。

この問題は画像キャッシュサーバーの仕様なのでしょうか？

**Comment by kyntk** [comment]
@MasatoMakino ご報告いただきありがとうございます。
ご不便をおかけして申し訳ございません。

調査いたします！

**Comment by kyntk** [comment]
@MasatoMakino 画像のプロキシをしているサーバーでエラーが発生していたため、修正いたしました。

現在は見れるようになっているはずなので､ご確認お願いします！

**Comment by MasatoMakino** [comment]
従来通り、外部サーバーへの画像指定が可能になったことを確認いたしました。

記事 https://qiita.com/masato_makino/items/dcc2ba13d28a5a6abab1 の [最新リビジョン](https://qiita.com/masato_makino/items/dcc2ba13d28a5a6abab1/revisions/8) で外部サーバー画像が正常に取得できています。

ご対応いただき誠にありがとうございます。
